dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/utils/TaskExecutionContextUtils.java

```diff
@@ -108,4 +111,21 @@ public static ResourceContext downloadResourcesIfNeeded(TaskChannel taskChannel,
         return resourceContext;
     }
 
+    public static void clearTaskInstanceWorkingDirectory(TaskExecutionContext taskExecutionContext) {
+        boolean isDevelopment = PropertyUtils.getBoolean(Constants.DEVELOPMENT_STATE, true);
+        if (!isDevelopment) {
```

Comment:
Move this in `finalizeTask` method, the utils only provide the delete logic here, then you can easilly test this.